requirements-dev.txt

```diff
@@ -1,5 +1,3 @@
-black
-flake8
```

Comment:
Suggest we add ruff here (unless it only ever runs on CI?)

Reply:
It runs in GitHub Actions and in pre-commit.ci, but seems useful to also have it in the dev requirements!